src/main.rs

```diff
@@ -0,0 +1,188 @@
+use std::error::Error;
+use std::ffi::{CStr, CString};
+use std::fmt::{Display, Formatter};
+use winit::event_loop::EventLoop;
+use winit::raw_window_handle::HasWindowHandle;
+use winit::window::{Window, WindowAttributes};
+
+use glutin::config::{Config, ConfigTemplateBuilder};
+use glutin::context::{ContextApi, ContextAttributesBuilder, NotCurrentContext, Version};
+use glutin::display::GetGlDisplay;
+use glutin::prelude::*;
+
+use glutin_winit::{DisplayBuilder, GlWindow};
+
+pub mod gl {
+    #![allow(clippy::all)]
+    include!(concat!(env!("OUT_DIR"), "/gl_bindings.rs"));
+
+    pub use Gles2 as Gl;
+}
+
+#[derive(Debug, Clone)]
+struct GLInfo {
+    driver: String,
+    vendor: String,
+    renderer: String,
+    version: String,
+    shading_language: String,
+}
+
+impl Display for GLInfo {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&format!(
+            "{} Vendor: {}\n\
+             Renderer: {}\n\
+             Version: {}\n\
+             Shading Language: {}",
+            self.driver, self.vendor, self.renderer, self.version, self.shading_language
+        ))
+    }
+}
+
+pub fn main() -> Result<(), Box<dyn Error>> {
+    let args: Vec<String> = std::env::args().collect();
+    if args.len() == 2 && (args[1] == "-h" || args[1] == "--help") {
+        println!("Usage: glinfo [-f filename]");
+        return Ok(());
+    }
+
+    let event_loop =
+        EventLoop::new().map_err(|err| format!("Failed to create event loop: {}", err))?;
```

Comment:
```suggestion
        EventLoop::new().map_err(|err| format!("Failed to create event loop: {err}"))?;
```
The same applies for `println`, as long as what you want to interpolate is in a variable (and not in a struct's field, for example).